src/main/scala/higherkindness/rules_scala/workers/common/LoggedReporter.scala

```diff
@@ -2,18 +2,18 @@ package higherkindness.rules_scala
 package workers.common
 
 import xsbti.{Logger, Problem}
-import sbt.internal.inc.{LoggedReporter => SbtLoggedReporter}
+import sbt.internal.inc.{LoggedReporter => SbtLoggedReporter, ProblemStringFormats}
 
 class LoggedReporter(logger: Logger, versionString: String) extends SbtLoggedReporter(0, logger) {
+  private val problemStringFormats = new ProblemStringFormats {}
+
   // Scala 3 has great error messages, let's leave those alone, but still add color to the Scala 2 messages
-  val shouldEnhanceMessage = if (versionString.startsWith("0.") || versionString.startsWith("3")) false else true
+  private val shouldEnhanceMessage = !versionString.startsWith("0.") && !versionString.startsWith("3")
+
+  private def doLog(problem: Problem, colorFunction: String => String): String = {
+    val formattedProblem = problemStringFormats.ProblemStringFormat.showLines(problem).mkString("\n")
 
-  def doLog(problem: Problem, colorFunc: Problem => String): String = {
-    if (shouldEnhanceMessage) {
-      colorFunc(problem)
-    } else {
-      problem.rendered().orElse(problem.toString)
-    }
+    if (shouldEnhanceMessage) colorFunction(formattedProblem) else formattedProblem
```

Comment:
Does this change how the Scala 3 messages look? Because those were very nice before.

Reply:
I don't think it should. Before, we were calling `Problem#rendered` and defaulting to `Problem#toString`. `ProblemStringFormats` also attempts to call `Problem#rendered`, but defaults a message that's much more detailed than `Problem#toString`
https://github.com/sbt/zinc/blob/develop/internal/zinc-compile-core/src/main/scala/sbt/internal/inc/javac/JavaErrorParser.scala#L59
https://github.com/sbt/zinc/blob/develop/internal/zinc-compile-core/src/main/scala/sbt/internal/inc/ProblemStringFormats.scala#L37

I think the niceness you're referring to is the product of `Problem#rendered`.

Reply:
I checked out this branch and a branch without this change. Confirmed they're still the same.